pathod/utils.py

```diff
@@ -69,7 +71,10 @@ def escape_unprintables(s):
     """
     s = s.replace("\r\n", "PATHOD_MARKER_RN")
     s = s.replace("\n", "PATHOD_MARKER_N")
-    s = inner_repr(s)
+    if six.PY2:
+        s = inner_repr(s)
+    else:
+        s = s.encode('unicode_escape').decode('ascii')
```

Comment:
So, in the code you linked, `s` is the thing pathod sends out on the wire,
which we escape here to preview it to the user. On the wire = bytes, so I
think `s` should always be bytes/BytesIO in the first place. Let's make
this bytes-only (and add a docstring indicating so)!

On Sun, 8 May 2016, 06:08 Shadab Zafar, <notifications@github.com> wrote:

> In pathod/utils.py
> <https://github.com/mitmproxy/mitmproxy/pull/1119#discussion_r62430307>:
>
> > @@ -69,7 +71,10 @@ def escape_unprintables(s):
> >      """
> >      s = s.replace("\r\n", "PATHOD_MARKER_RN")
> >      s = s.replace("\n", "PATHOD_MARKER_N")
> > -    s = inner_repr(s)
> > +    if six.PY2:
> > +        s = inner_repr(s)
> > +    else:
> > +        s = s.encode('unicode_escape').decode('ascii')
>
> escape_unprintables is used here
> <https://github.com/dufferzafar/mitmproxy/blob/pathod-port/pathod/app.py#L169>
> .
>
> The input to it is bytes in Py2, and unicode in Py3.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/mitmproxy/mitmproxy/pull/1119/files/50bbe91691c283e0f181ea81084bfa15def9be72#r62430307>
>